.dotnet.azure/src/Custom/Assistants/AzureAssistantClient.Protocol.cs

```diff
@@ -29,13 +29,13 @@ public override ClientResult CreateAssistant(BinaryContent content, RequestOptio
 
     public override IAsyncEnumerable<ClientResult> GetAssistantsAsync(int? limit, string order, string after, string before, RequestOptions options)
     {
-        AzureAssistantsPageEnumerator enumerator = new(Pipeline, _endpoint, limit, order, after, before, _apiVersion, options);
+        using AzureAssistantsPageEnumerator enumerator = new(Pipeline, _endpoint, limit, order, after, before, _apiVersion, options);
```

Comment:
Have we functionally validated with the `using` statement and/or do we have other cases where we use `using` and it's known working?

I'm suspicious that this scoping may cause the enumerator to be disposed immediately upon returning from this method, which would leave the returned page collection having an already-disposed reference it's trying to use.

Reply:
you are probably correct looking at how the enumerator is stored in the PageCollectionHelpers.CreateAsync functions. Do we have a test that enumerates this? I feel like it should break every time.